### PR TITLE
refactor: rename `project` -> `inject`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/7163

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Also needs changes in Lowering, I think

**Comment by jaschdoc** [comment]
Looks like a Parser error:
```
- Test.Term.Lit.flix *** FAILED ***
  Unable to compile. Failed with: 1 errors.
  
  -- Parse Error -------------------------------------------------- main/test/flix/Test.Term.Lit.flix
  
  >> Parse Error: Invalid input "inj", expected ':', "->", '.', ::, :::, "without", Op, '`', <+>, *, /, +, -, <=, >=, <, >, ==, != or <=> (line 10, column 20):
          solve f, p inject R
                     ^
  
  Syntactic Context: OtherExpr. (FlixSuite.scala:69)
               
```

**Comment by magnus-madsen** [comment]
https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Parser.scala#L1231

**Comment by magnus-madsen** [comment]
We may also want to do some renaming in the ASTs.

File: examples/larger-examples/datalog/connect-graph.flix [hunk 1]
```diff
@@ -46,7 +46,7 @@ pub def connectedComponentRep(g: Graph[node]): #{ ComponentRep(node, node) | r }
         // `rep` is the highest node of the component.
         ComponentRep(n, rep) :- Reachable(n, rep), not ReachUp(rep).
     };
-    solve nodes, edges, reachability project ComponentRep
+    solve nodes, edges, reachability inject ComponentRep
```
Comment (JonathanStarup) [inline examples/larger-examples/datalog/connect-graph.flix:49]: shouldn't solve ... project ... still use project?

Comment (magnus-madsen) [reply examples/larger-examples/datalog/connect-graph.flix:49]: We renamed project to inject.

Comment (magnus-madsen) [reply examples/larger-examples/datalog/connect-graph.flix:49]: Gosh. You are right

Comment (magnus-madsen) [reply examples/larger-examples/datalog/connect-graph.flix:49]: @jaschdoc As Jonathan points out. For solve it should still be `solve ... project ...`.

Comment (jaschdoc) [reply examples/larger-examples/datalog/connect-graph.flix:49]: Ah ok I see that makes more sense.

Comment (magnus-madsen) [reply examples/larger-examples/datalog/connect-graph.flix:49]: Its a question of direction.

**Pull request closed without merging**